api/server/router/image/image_routes.go

```diff
@@ -67,7 +71,61 @@ func (ir *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrit
 		progressErr = ir.backend.PullImage(ctx, img, tag, platform, metaHeaders, authConfig, output)
 	} else { // import
 		src := r.Form.Get("fromSrc")
-		progressErr = ir.backend.ImportImage(ctx, src, repo, platform, tag, message, r.Body, output, r.Form["changes"])
+
+		var ref reference.Named
+		if repo != "" {
+			var err error
+			ref, err = reference.ParseNormalizedNamed(repo)
+			if err != nil {
+				return errdefs.InvalidParameter(err)
+			}
+			if _, isCanonical := ref.(reference.Canonical); isCanonical {
+				return errdefs.InvalidParameter(errors.New("cannot import digest reference"))
+			}
+
+			if tag != "" {
+				ref, err = reference.WithTag(ref, tag)
+				if err != nil {
+					return errdefs.InvalidParameter(err)
+				}
+			} else {
+				ref = reference.TagNameOnly(ref)
+			}
+		}
+
+		if len(comment) == 0 {
+			comment = "Imported from " + src
+		}
+
+		var layerReader io.ReadCloser
+		defer r.Body.Close()
+		if src == "-" {
+			layerReader = r.Body
+		} else {
+			if len(strings.Split(src, "://")) == 1 {
```

Comment:
TIL we can import remote tars.

Reply:
LOL. I completely forgot about that too 😂